src/containers/singleEvent/index.tsx

```diff
@@ -86,31 +87,47 @@ const GrayButton = styled(StyledButton)`
 
 interface SingleEventProps {
   readonly events: EventsReducerState['upcomingEvents'];
+  readonly eventAnnouncements: EventAnnouncementsReducerState['eventAnnouncements'];
 }
 
 interface SingleEventParams {
   id: string;
 }
 
-const SingleEvent: React.FC<SingleEventProps> = ({ events }) => {
+const SingleEvent: React.FC<SingleEventProps> = ({
+  events,
+  eventAnnouncements,
+}) => {
   const dispatch = useDispatch();
+  const id = Number(useParams<SingleEventParams>().id);
+
   useEffect(() => {
-    if (asyncRequestIsNotStarted(events) || asyncRequestIsFailed(events)) {
+    if (
+      asyncRequestIsNotStarted(events) ||
+      asyncRequestIsFailed(events) ||
+      asyncRequestIsNotStarted(eventAnnouncements) ||
+      asyncRequestIsFailed(eventAnnouncements)
+    ) {
       dispatch(getUpcomingEvents());
+      dispatch(getEventAnnouncements(id));
     }
-  }, [dispatch, events]);
+  }, [dispatch, events, eventAnnouncements, id]);
+
+  useEffect(() => {
+    dispatch(getUpcomingEvents());
+    dispatch(getEventAnnouncements(id));
+  }, [dispatch, id]);
```

Comment:
❓ Why is the useEffect and dispatch duplicated? I think we'll make two calls here if the first useEffect is ran before this one.

Reply:
In the other one, once the async request is complete, it never fetches the announcements again. So the same announcements show up for all the events. So this useEffect makes sure we reload the announcements every time the id (event) changes. I think I just remove the other one though, because that one only really works on the initial time, whereas this one works every time.

Reply:
If that didn't make any sense let me know